cilium/bpf_metadata.cc

```diff
@@ -349,7 +349,8 @@ const PolicyInstanceConstSharedPtr Config::getPolicy(const std::string& pod_ip)
   return policy;
 }
 
-Cilium::SocketOptionSharedPtr Config::getMetadata(Network::ConnectionSocket& socket) {
+Cilium::BpfMetadata::SocketMetadataSharedPtr
+Config::extractSocketMetadata(Network::ConnectionSocket& socket) {
```

Comment:
The intermediate result should not need to be a (shared) pointer, should use `absl::optional` instead. Upstream code base has plenty of examples for it :-)

Reply:
thanks for the hint - refactored